explorer/interpreter/type_checker.cpp

```diff
@@ -674,11 +690,10 @@ auto TypeChecker::ArgumentDeduction(
                    << var_type.binding() << ":\n  " << *it->second << "\n  "
                    << *arg;
           }
+          return Success();
         }
-      } else {
-        return handle_non_deduced_type();
       }
-      return Success();
+      return handle_non_deduced_type();
```

Comment:
Do we still need this change? If the binding for the `VariableType` is in `deduced`, then we're not in the "non-deduced type" case so I would think we don't need to do this.

Reply:
you are right